core/src/main/scala/cats/Trivial.scala

```diff
@@ -1,23 +0,0 @@
-package cats
-
-/**
- * The "Unit type class".  The only instance of `Trivial` is given by
- * `Trivial.manifest`, and this instance is guaranteed to be in the
- * implicit scope.  Several convenience type aliases are provided in
- * companion object, covering a few common use cases and avoiding the
- * need for unnecessary lambdas (e.g. if you want a trivial type class
- * instance for a type constructor, you should use `Trivial.PH1`).
- */
-sealed trait Trivial
-
-object Trivial {
```

Comment:
I understand that as a result of this PR `Trivial` is no longer needed in cats, but it is still a very useful type constraint. Is there a good reason to remove `Trivial`?